apis/addons/v1alpha1/addons_types.go

```diff
@@ -14,9 +21,17 @@ type AddonStatus struct {
 	// Conditions is a list of status conditions ths object is in.
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 	// Human readable status aggregated from conditions.
-	Phase string `json:"phase,omitempty"`
+	Phase AddonPhase `json:"phase,omitempty"`
 }
 
+type AddonPhase string
+
+// These are the valid phases of an Addon
+const (
+	AddonNamespacesPending AddonPhase = "AddonNamespacesPending"
+	AddonNamespacesReady   AddonPhase = "AddonNamespacesReady"
```

Comment:
These phases are too granular.
In the end that phase is just a simple help to get `kubectl/oc` to print something in a Status column and there is little need to update them in very short succession. I would concentrate on only two transition states and errors:
- "Pending" - not done anything, should be the default via `// +kubebuilder:default="Pending"`
- "Ready" - everything ready
- "ErrorNamespaceCollision" - Error reconciling namespaces
- "ErrorXXX" - Other error status